pymongo/message.py

```diff
@@ -399,7 +394,7 @@ def as_command(
         # Support auto encryption
         client = self.client
         if client._encrypter and not client._encrypter._bypass_auto_encryption:
-            cmd = cast(SON[str, Any], client._encrypter.encrypt(self.db, cmd, self.codec_options))
+            cmd = cast(dict[str, Any], client._encrypter.encrypt(self.db, cmd, self.codec_options))
```

Comment:
Let's change _encrypter.encrypt's return type to dict so we can remove the redundant casts here.